templates/_helpers.tpl

```diff
@@ -25,6 +25,19 @@ If release name contains chart name it will be used as a full name.
 {{- end }}
 {{- end }}
 
+{{/*
+Construct the namespace for all namespaced resources
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+Preserve the default behavior of the Release namespace if no override is provided
+*/}}
+{{- define "harbor.namespace" -}}
```

Comment:
this is much easier:

{{- default .Release.Namespace .Values.namespaceOverride | trunc 63 | trimSuffix "-" -}}

Reply:
Compelted.